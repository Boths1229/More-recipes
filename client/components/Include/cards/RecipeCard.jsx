import React from 'react';
import PropTypes, { any } from 'prop-types';
import { Link } from 'react-router-dom';
import RecipeCardFooter from './RecipeCardFooter';

/**
 * @description recipe card
 *
 * @param {props} props
 *
 * @returns {JSX} return JSX
 */

const RecipeCard = (props) => {
  const { recipeData, favRecipeData } = props;
  return (
    <div className="col-sm-6 col-md-4 pt-4 mb-5">
      <div className="card">
        <img
          className="card-img-top img-max"
          src={recipeData.image || favRecipeData.image}
          alt=""
        />
        <div className="card-body">
          <Link to={`/recipes/${recipeData.id || favRecipeData.id}`}>
            <h4 className="card-title">{recipeData.name || favRecipeData.name}</h4>
          </Link>
          <p className="card-text pt-2">
            {recipeData.description || favRecipeData.description}
          </p>
        </div>
        <RecipeCardFooter
          data={props}
        />
      </div>
    </div>
  );
};

RecipeCard.propTypes = {
  recipeData: PropTypes.objectOf(any).isRequired
};

export default RecipeCard;
