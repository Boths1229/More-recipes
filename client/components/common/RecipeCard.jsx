import React from 'react';
import PropTypes, { any } from 'prop-types';
import RecipeCardFooter from '../Include/RecipeCardFooter';

/**
 * @description recipe card
 *
 * @param {props} props
 *
 * @returns {JSX} return JSX
 */

const RecipeCard = (props) => {
  const { recipeData } = props;
  const {
    name, description, image
  } = recipeData;
  return (
    <div className="col-sm-6 col-md-4 pt-4 mb-5">
      <div className="card">
        <img className="card-img-top img-max" src={image} alt="" />
        <div className="card-body">
          <a href="recipeDetail.html">
            <h4 className="card-title">{name}</h4>
          </a>
          <p className="card-text pt-2">
            {description}
          </p>
        </div>
        <RecipeCardFooter
          data={props}
        />
      </div>
    </div>
  );
};

RecipeCard.propTypes = {
  recipeData: PropTypes.objectOf(any).isRequired
};

export default RecipeCard;
