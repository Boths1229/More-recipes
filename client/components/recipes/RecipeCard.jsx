import React from 'react';
import PropTypes, { any } from 'prop-types';

/**
 * @description recipe card
 *
 * @param {props} props
 *
 * @returns {JSX} return JSX
 */

const RecipeCard = (props) => {
  const {
    recipeData, upvoteRecipe, downvoteRecipe, favoriteRecipe
  } = props;
  const {
    id,
    name,
    description,
    upvotes,
    image,
    downvotes
  } = recipeData;

  return (
    <div className="col-sm-6 col-md-4 pt-4 mb-5">
      <div className="card">
        <img className="card-img-top img-max" src={image} alt="" />
        <div className="card-body">
          <a href="recipeDetail.html">
            <h4 className="card-title">{name}</h4>
          </a>
          <p className="card-text pt-2">
            {description}
          </p>
        </div>
        <ul className="list-group list-group-flush">
          <li className="list-group-item card-tile">
            <i
              className="fa fa-heart fa-2x favorite-btn"
              aria-hidden="true"
              onClick={() => favoriteRecipe(id)}
            />
            <i
              className="fa fa-thumbs-up fa-2x ml-5 favorite-btn"
              aria-hidden="true"
              onClick={() => upvoteRecipe(id)}
            >
              <span className="small pr-2">{upvotes}
              </span>
            </i>
            <i
              className="fa fa-thumbs-down fa-2x favorite-btn"
              aria-hidden="true"
              onClick={() => downvoteRecipe(id)}
            >
              <span className="small pr-2">
                {downvotes}
              </span>
            </i>
          </li>
        </ul>
      </div>
    </div>
  );
};

RecipeCard.propTypes = {
  recipeData: PropTypes.objectOf(any).isRequired,
  upvoteRecipe: PropTypes.func.isRequired,
  downvoteRecipe: PropTypes.func.isRequired,
  favoriteRecipe: PropTypes.func.isRequired
};

export default RecipeCard;
