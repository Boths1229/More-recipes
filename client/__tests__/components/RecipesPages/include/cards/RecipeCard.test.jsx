import React from 'react';
import { shallow } from 'enzyme';
import RecipeCard from '../../../../../components/Include/cards/RecipeCard';

describe('Recipe card component', () => {
  it('matches snapshot', () => {
    const props = {
      recipe: {}
    };
    const output = shallow(<RecipeCard {...props} />);
    expect(output).toMatchSnapshot();
  });
});
