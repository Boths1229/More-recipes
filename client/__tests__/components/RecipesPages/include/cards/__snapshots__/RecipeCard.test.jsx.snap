// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe card component matches snapshot 1`] = `
<div
  className="col-sm-6 col-md-6 col-lg-4 pt-4 mb-5"
>
  <div
    className="card card-height"
  >
    <img
      className="card-img-top img-max"
    />
    <div
      className="card-body word-warp"
    >
      <Link
        replace={false}
        to="/recipes/undefined"
      >
        <h4
          className="card-title text-center"
        />
      </Link>
    </div>
    <RecipeCardFooter
      recipe={Object {}}
    />
  </div>
</div>
`;
